jax/_src/pjit.py

```diff
@@ -844,36 +903,36 @@ def _create_pjit_jaxpr(fun, global_in_avals, debug_info, out_paths):
 
 @lru_cache(maxsize=4096)
 def _check_and_canonicalize_out_shardings(
-    out_shardings_thunk, out_tree, global_out_avals):
+    out_shardings_thunk, out_tree, out_type):
   orig_out_shardings = out_shardings_thunk()
   # TODO(yashkatariya): Remove the if branch and fix flatten_axis_resources
   # instead. This condition exists because flatten_axis_resources passes in an
   # `object()` while unflattening which breaks assertion is user defined
   # pytrees (which shouldn't exist but they do).
   if (_is_unspecified(orig_out_shardings) or
       isinstance(orig_out_shardings, XLACompatibleSharding)):
-    out_shardings_flat = (orig_out_shardings,) * len(global_out_avals)
+    out_shardings_flat = (orig_out_shardings,) * len(out_type)
   else:
     out_shardings_flat = flatten_axis_resources(
         "pjit out_shardings", out_tree(), orig_out_shardings,
         tupled_args=False)
 
-  pjit_check_aval_sharding(out_shardings_flat, global_out_avals, "pjit outputs",
+  pjit_check_aval_sharding(out_shardings_flat, out_type, "pjit outputs",
```

Comment:
Should this be guarded with jax.config.dynamic_shapes too as you are guarding in _process_in_axis_resources?

Reply:
Sorry, what do you mean by "this" here?

Reply:
Ohh sorry.. I meant `pjit_check_aval_sharding` line.

Should it be guarded with `if not config.jax_dynamic_shapes`?

Reply:
(`out_type` is just the new variable name for what was previously called `global_out_avals`)

Reply:
Ohh I see! So out_type won't contain extra stuff like in_type does when dynamic_shapes is on?

Reply:
Ohh nvm.. I read the _pjit_jaxpr code again. 

Then why rename this? Why not just call it global_out_avals? (given that it is avals)

The out_type confused me since I associated it with in_type

Reply:
Ah yes, thank you!